bottlecap/src/config/mod.rs

```diff
@@ -93,6 +97,46 @@ fn log_failover_reason(reason: &str) {
     println!("{{\"DD_EXTENSION_FAILOVER_REASON\":\"{reason}\"}}");
 }
 
+fn failsover(figment: &Figment) -> Result<(), ConfigError> {
+    let failover_config: FailoverConfig = match figment.extract() {
+        Ok(failover_config) => failover_config,
+        Err(err) => {
+            println!("Failed to parse Datadog config: {err}");
+            return Err(ConfigError::ParseError(err.to_string()));
+        }
+    };
+
+    let opted_in = match failover_config.extension_version.as_deref() {
+        Some("next") => true,
+        // Only log when the field is present but its not "next"
+        Some(_) => {
+            log_failover_reason("extension_version");
+            false
+        }
+        _ => false,
+    };
+
+    if !opted_in {
+        return Err(ConfigError::UnsupportedField(
+            "extension_version".to_string(),
+        ));
+    }
+
+    if failover_config.serverless_appsec_enabled || failover_config.appsec_enabled {
+        log_failover_reason("appsec_enabled");
+        return Err(ConfigError::UnsupportedField("appsec_enabled".to_string()));
+    }
+
+    if failover_config.profiling_enabled {
+        log_failover_reason("profiling_enabled");
+        return Err(ConfigError::UnsupportedField(
+            "profiling_enabled".to_string(),
+        ));
+    }
+
```

Comment:
Let's add a failover condition for `AWS_LAMBDA_WRAPPER_SCRIPT` as well. That's required for Java and DotNet and would probably catch most universal instrumentation users who may opt in without reading the docs fully.

Reply:
Sg